Redeclaration of '_t2' while twice dl usage

### Describe the bug

I've wrote two dll wrappers with dl lib. When I tried to import my libraries, I encountered an error: **redeclaration of '_t2'**

### Reproduction Steps

In two different files

module1.v, module2.v

```v
import dl

const handle = dl.open_opt(utils.pub_get_dl_path(), dl.rtld_lazy) or { panic(err) }

// some dl.sym_opt methods

```

main.v

```v
module main

import module1
import module2

fn main() {
    module1.method1()
    module2.method2()
}

### Expected Behavior

Correct compilation

### Current Behavior

C:/Users/nikit/AppData/Local/Temp/v_0/main.01JQPWE9018SJQXNF4PR3Z5A1X.tmp.c:615: warning: WINVER redefined
C:/Users/nikit/AppData/Local/Temp/v_0/main.01JQPWE9018SJQXNF4PR3Z5A1X.tmp.c:6398: warning: implicit declaration of function 'tcc_backtrace'
C:/Users/nikit/AppData/Local/Temp/v_0/main.01JQPWE9018SJQXNF4PR3Z5A1X.tmp.c:13050: error: redeclaration of '_t2'
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

0.4.10 2641a55.1b52538

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.10 2641a55.1b52538
|:-------------------|:-------------------
|OS                  |windows, Њ ©Єа®б®дв Windows 10 Pro 19045 64-а §ап¤­ п
|Processor           |12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz  
|Memory              |48.18GB/63.92GB
|                    |
|V executable        |C:\Users\nikit\v\v.exe
|V last modified time|2025-03-24 20:52:19
|                    |
|V home dir          |OK, value: C:\Users\nikit\v
|VMODULES            |OK, value: C:\Users\nikit\\.vmodules
|VTMP                |OK, value: C:\Users\nikit\AppData\Local\Temp\v_0
|Current working dir |OK, value: C:\Users\nikit\CExploringLab\PROJECTS\tiny-gpu-watcher
|                    |
|Git version         |git version 2.47.1.windows.2
|V git status        |weekly.2024.03-2641-g1b52538d (31 commit(s) behind V master)
|.git/config present |true
|                    |
|cc version          |cc (Rev6, Built by MSYS2 project) 13.1.0
|gcc version         |gcc (Rev6, Built by MSYS2 project) 13.1.0
|clang version       |N/A
|msvc version        |N/A
|tcc version         |tcc version 0.9.27 (x86_64 Windows)
|tcc git status      |thirdparty-windows-amd64 b425ac82
|emcc version        |N/A
|glibc version       |ldd (cygwin) 2.5.0

--- a/vlib/v/gen/c/consts_and_globals.v
+++ b/vlib/v/gen/c/consts_and_globals.v
@@ -129,7 +129,9 @@ fn (mut g Gen) const_decl(node ast.ConstDecl) {
 							continue
 						}
 					}
-					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false)
+					should_surround := field.expr.expr is ast.CallExpr
+						&& field.expr.expr.or_block.kind != .absent
+					g.const_decl_init_later(field.mod, name, field.expr, field.typ, should_surround)
 				} else if field.expr is ast.InfixExpr {
 					mut has_unwrap_opt_res := false
 					if field.expr.left is ast.CallExpr {

--- a/vlib/v/gen/c/testdata/const_init_or_block.c.must_have
+++ b/vlib/v/gen/c/testdata/const_init_or_block.c.must_have
@@ -1,0 +1,11 @@
+{
+_option_bool _t2 = main__t1();
+if (_t2.state != 0) {
+IError err = _t2.err;
+_v_panic(IError_str(err));
+VUNREACHABLE();
+;
+}
+
+_const_main__barz = (((*(bool*)_t2.data)));
+}

--- a/vlib/v/gen/c/testdata/const_init_or_block.vv
+++ b/vlib/v/gen/c/testdata/const_init_or_block.vv
@@ -1,0 +1,9 @@
+// vtest vflags: -no-skip-unused
+fn t1() ?bool {
+	return true
+}
+
+type FooBar = bool
+
+const bar = FooBar(t1() or { panic(err) })
+const barz = FooBar(t1() or { panic(err) })

